examples/timeout/main_e2e_test.go

```diff
@@ -0,0 +1,70 @@
+//go:build e2e
+
+package main
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/hatchet-dev/hatchet/internal/testutils"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+)
+
+func TestTimeout(t *testing.T) {
+	testutils.Prepare(t)
+
+	tests := []struct {
+		name string
+		job  worker.WorkflowJob
+	}{
+		{
+			name: "step timeout",
+			job: worker.WorkflowJob{
+				Name:        "timeout",
+				Description: "timeout",
+				Steps: []*worker.WorkflowStep{
+					worker.Fn(func(ctx worker.HatchetContext) (result *stepOneOutput, err error) {
+						time.Sleep(time.Second * 60)
```

Comment:
I think it'd be better if we made sure the timeout signal propagates to the step over `ctx.Done()`, rather than just sleeping

Reply:
Good point, I thought we already tested this in the cancellation but indeed it's not quite the same thing. I'll adapt.